low-level/framework/utils/salt_util.py

```diff
@@ -104,18 +130,23 @@ def get_consul_port(self):
         consul_port = '8500'
         # Read consul port from sspl.sls
         try:
-            SALT_CMD = f"sudo salt-call pillar.get sspl:DATASTORE:consul_port --output=newline_values_only"
-            subout = subprocess.Popen(SALT_CMD, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
-            result = subout.stdout.readlines()
-            if result == [] or result == "":
-                logger.warning("Consul Port fetch failed, using 8500 to connect.")
+            if self.pillar_info is not None:
+                data_store_config = \
+                    self.pillar_info[self.node_id][self.SSPL_KEY][self.DATASTORE_KEY]
+                if data_store_config is not None:
+                    consul_vip = data_store_config['consul_port']
             else:
-                consul_port = result[0].decode().rstrip('\n')
-        except Exception as e:
-            logger.warning(f"Consul Port read failed, using 8500 to connect : {e}")
+                logger.warning(f'sspl_constants, Fail to read consul VIP port with \
+                                 this error: {_err}, Assuming localhost: {consul_port} connection')
+        except Exception as err:
+            logger.warning(f'sspl_constants, Fail to read consul VIP with \
```

Comment:
print both vip and port.

Reply:
port is getting fetched in another function. So, at this time, we don't have port info yet.